### PR TITLE
fix: better caching

### LINKED ISSUE #25: Bot crashes if a message is sent in a channel made after the bot is created
## Error
```
Error: Cannot reply to a message that does not have a channel.
at Message.reply (/usr/src/app/node_modules/ailurus/src/structures/Message.ts:24:28)
at PingCommand.exec (/usr/src/app/src/commands/general/ping.ts:15:18)
at CommandHandler.handle (/usr/src/app/src/CommandHandler.ts:80:12)
at Client.<anonymous> (/usr/src/app/src/CommandHandler.ts:67:9)
at Client.emit (node:events:390:28)
at WebSocket.<anonymous> (/usr/src/app/node_modules/ailurus/src/client/ws/Gateway.ts:82:20)
at WebSocket.onMessage (/usr/src/app/node_modules/ws/lib/event-target.js:199:18)
at WebSocket.emit (node:events:390:28)
at Receiver.receiverOnMessage (/usr/src/app/node_modules/ws/lib/websocket.js:1137:20)
at Receiver.emit (node:events:390:28)
```

We need a live-updating cache sort of system.

### PR DESCRIPTION
**Pull request by Nciklol**
- Fixes #25 

- Caches guilds properly
- Adds support for `GUILD_REMOVE`, `CHANNEL_CREATE`, and `CHANNEL_DELETE`

### CODE REVIEW

**Review by twisttaan — APPROVED** [review]
Looks good, nice!

File: package.json [hunk 1]
```diff
@@ -43,10 +43,10 @@
 		"update": "yarn up \"*\" -i"
 	},
 	"dependencies": {
+		"discord-api-types": "^0.31.2",
 		"dotenv": "^16.0.0",
 		"undici": "^5.0.0",
-		"ws": "^8.5.0",
-		"discord-api-types": "^0.31.2"
+		"ws": "^8.5.0"
```
Comment (Kurpp) [inline package.json:49]: Could you revert these changes? Nothing was changed so it seems unnecessary to shuffle them around

**Review by Kurpp — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (twisttaan) [reply package.json:49]: I'm pretty sure some lint got run and put in some ordering

Comment (Nciklol) [reply package.json:49]: Yeah I wasn't sure why these committed, good catch.

**Review by Kurpp — APPROVED** [review]
LGTM
